wasm-test-app/src/Utils.tsx

```diff
@@ -0,0 +1,39 @@
+export function readUploadedFileAsText(file: File) {
+    return new Promise<Uint8ClampedArray>((resolve, reject) => {
+        const reader = new FileReader();
+
+        reader.onload = event => {
+            if (event.target && event.target.result) {
+                const arrayBuffer = event.target.result as ArrayBuffer;
```

Comment:
I try to avoid using `as` as well. Can you pass a generic type to `FileReader`? If not I like to opt for building my own typed version the API and putting it in a `utils` dir: https://www.newline.co/@bespoyasov/how-to-use-fetch-with-typescript--a81ac257#wrapper-module